test/e2e/common/autoscaling_utils.go

```diff
@@ -554,3 +553,11 @@ func CreateCPUHorizontalPodAutoscaler(rc *ResourceConsumer, cpu, minReplicas, ma
 func DeleteHorizontalPodAutoscaler(rc *ResourceConsumer, autoscalerName string) {
 	rc.clientSet.AutoscalingV1().HorizontalPodAutoscalers(rc.nsName).Delete(autoscalerName, nil)
 }
+
+// runDeployment runs a delopyment with the specified config.
+func runDeployment(config testutils.DeploymentConfig) error {
```

Comment:
nit: I don't think having this as a separate function is needed.  Can we just set the `NodeDumpFunc` and `ContainerDumpFunc` when we create the `testutils.DeploymentConfig` above, and do `framework.ExpectNoError(testutils.RunDeployment(dpConfig)))`

Reply:
@dashpole Thanks for your review, that is a nice idea.
I will do that with another follow-up PR.